pkg/supervisor/supervisor_test.go

```diff
@@ -15,10 +15,19 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// Helper for building a GetLedgerLatency func.
+func mockGetLedgerLatency(duration time.Duration, err error) func(ctx context.Context) (time.Duration, error) {
```

Comment:
We never use the `err` argument in any of the call sites.  An argument could be made to just remove the parameter here.

Reply:
Good point! I was thinking of adding tests for what happens when an `err` is returned, but never got around to that.